### ISSUE TITLE
mingw-w64-make fails to build with error 'long int' vs 'long unsigned int'

### ISSUE THREAD

**Issue by emmenlau** [Alexpux/MINGW-packages#2192]
I fail to compile latest `mingw-w64-make` against the latest toolchain on `x86_64` (package sources from today). The build fails with multiple errors. Some errors are related to '%ld' type:
```
In file included from ../make/function.c:17:0:
../make/function.c: In function 'windows32_openpipe':
../make/function.c:1526:17: error: format '%ld' expects argument of type 'long int', but argument 4 has type 'DWORD {aka long unsigned int}' [-Werror=format=]
               _("windows32_openpipe: DuplicateHandle(In) failed (e=%ld)\n"), e);
                 ^
```
And there are also errors about function return value casts:
```
../make/function.c:1530:12: error: cast from function call of type 'intptr_t {aka long long int}' to non-matching type 'void *' [-Werror=bad-function-cast]
   tmpErr = (HANDLE)_get_osfhandle (errfd);
            ^
```
Find the build log here: http://data.biodataanalysis.de/tmp/buildlogs/1487611087-mingw-w64-make-buildlog.txt

**Comment by mati865** [comment]
This version will probably not build with GCC 6 easily. You could try changing from git to release version 4.2.1 https://ftp.gnu.org/gnu/make/

**Comment by emmenlau** [comment]
It looks like there are many places where `DWORD` is formatted with `%ld` instead of `%lu`. I checked also in 4.2.1 release, I think its the same issue. Should I just fix it everywhere in the code, make a patch and supply it upstream? Do you know are patches accepted easily upstream?

**Comment by mati865** [comment]
You could try removing `maintMakefile` file (releases don't have it). It is included by [configure ](http://git.savannah.gnu.org/cgit/make.git/tree/configure.ac#n443) and contains [flags converting warnings to errors](http://git.savannah.gnu.org/cgit/make.git/tree/maintMakefile#n23).
It's normal practice for git versions to do stricter checks.

> Do you know are patches accepted easily upstream?

I have no idea.

**Comment by emmenlau** [comment]
Ok thanks again @mati865 , I've followed your recommendation and downgraded make to 4.2.1 release. With tiny changes to the build process it works mostly out of the box. Before I tried to fix the build from the upstream HEAD but there where issues I could not easily fix.

I've created a PR here: https://github.com/Alexpux/MINGW-packages/pull/2195 But will this PR be accepted? It fixes the broken build... maybe it can be accepted for the time being, and when someone gets around to fix the upstream problems we can switch to git again?
